avro-converter/src/main/java/io/confluent/copycat/avro/AvroConverter.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.copycat.avro;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.serializers.AbstractKafkaAvroDeserializer;
+import io.confluent.kafka.serializers.AbstractKafkaAvroSerializer;
+import io.confluent.kafka.serializers.NonRecordContainer;
+import org.apache.avro.generic.GenericContainer;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.copycat.data.Schema;
+import org.apache.kafka.copycat.data.SchemaAndValue;
+import org.apache.kafka.copycat.errors.DataException;
+import org.apache.kafka.copycat.storage.Converter;
+
+import java.util.Map;
+
+/**
+ * Implementation of Converter that uses Avro schemas and objects.
+ */
+public class AvroConverter implements Converter {
+  private Serializer serializer;
+  private Deserializer deserializer;
+
+  private boolean isKey;
+
+  public AvroConverter() {
+    serializer = new Serializer();
+    deserializer = new Deserializer();
+  }
+
+  public AvroConverter(SchemaRegistryClient client) {
```

Comment:
Guava has an annotation for this (VisibleForTesting), but I don't think there's one built-in or in any of the deps we already use. I've just used EasyMock/PowerMock in other code (e.g. the Copycat packages in Kafka) to avoid this messiness. I stuck with the current convention here, but we could also just start using that approach in this code.